chromium_src/base/feature_list.cc

```diff
@@ -0,0 +1,48 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "base/feature_list.h"
+#include "base/feature_override.h"
+#include "base/no_destructor.h"
+
+namespace base {
+namespace {
+
+using FeatureDefaultStateOverrides =
+    base::flat_map<const Feature*, FeatureState>;
+
+FeatureDefaultStateOverrides& GetFeatureDefaultStateOverrides() {
+  static base::NoDestructor<FeatureDefaultStateOverrides> overrides;
+  return *overrides;
+}
+
+FeatureState GetDefaultOrOverriddenState(const Feature& feature) {
+  const auto& overrides = GetFeatureDefaultStateOverrides();
+  auto override_it = overrides.find(&feature);
+  return override_it == overrides.end() ? feature.default_state
+                                        : override_it->second;
+}
+
+}  // namespace
+
+namespace internal {
+
+FeatureDefaultStateOverride::FeatureDefaultStateOverride(
+    const base::Feature& feature,
+    FeatureState default_state) {
+  const auto& inserted =
+      GetFeatureDefaultStateOverrides().emplace(&feature, default_state);
+  DCHECK(inserted.second) << "Feature " << feature.name
+                          << "has already been overridden";
+}
+
+}  // namespace internal
+}  // namespace base
+
+#define default_state name&& GetDefaultOrOverriddenState(feature)
```

Comment:
Nit: missing space before `&&`. Also, maybe a brief comment mentioning that this is meant as a way to "expand" the few `feature.default_state == FEATURE_ENABLED_BY_DEFAULT` statements would be useful for the reader.

Reply:
actually this is how clang format does it. left as is.

Reply:
Weird but ok :shrug: